src/deferred.js

```diff
@@ -103,7 +103,10 @@ jQuery.extend( {
 						fns = null;
 					} ).promise();
 				},
-				then: function( onFulfilled, onRejected, onProgress ) {
+
+                // The onException is there to manage that an exception is thrown
+                // inside the onFullfilled without being catched.
+				then: function( onFulfilled, onRejected, onProgress, onException ) {
```

Comment:
We avoid deviating from the standards as well as adding new undocumented parameters to publicly available methods. Also, the current Deferreds API should be enough to achieve the goals of this PR. All we need is to automatically attach a special `.catch` handler to the `readyList.then( fn )` promise in the `core/ready.js` file.